src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitElement = fromFormat[3];
+  const joinElement = toFormat[3];
+  const newDate = date.split(splitElement);
+
+  let formatAgan = [];
+
+  let year = formatAgan[0];
+  let month = formatAgan[1];
+  let day = formatAgan[2];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      year = newDate[i];
+    }
+
+    if (fromFormat[i] === 'YYYY') {
+      year = newDate[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = newDate[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      day = newDate[i];
+    }
+  }
+
+  if (toFormat.includes('YYYY') && year.length === 2 && year < 30) {
+    year = `${20}` + `${year}`;
+  }
+
+  if (toFormat.includes('YYYY') && year.length === 2 && year >= 30) {
+    year = `${19}` + `${year}`;
+  }
+
+  if (toFormat.includes('YY')) {
```

Comment:
The condition `toFormat.includes('YY')` will always return true if 'YY' is anywhere in the `toFormat` array, even if 'YY' is not the desired format for the year part. You should check the specific index of the year format in `toFormat` array (similar to what you did when reading `fromFormat`) to determine if you need to convert 'YYYY' to 'YY'.